File: scratch/inspect_deltas.py
import os
import requests

baseUrl = os.getenv('NEXT_PUBLIC_QFIELD_API_URL') or 'http://localhost:8100'
qfield_project_id = '7a55babe-8909-4554-8dd7-9bc03f7a1b9b'

username = os.getenv('QFIELD_ADMIN_USER') or 'admin'
password = os.getenv('QFIELD_ADMIN_PASS') or 'admin'

print("Connecting to QFieldCloud...")
res = requests.post(f"{baseUrl}/api/v1/auth/login/", json={"username": username, "password": password})
if not res.ok:
    print(f"Auth failed: {res.status_code} - {res.text}")
    exit(1)

token = res.json().get('token') or res.json().get('access_token')
headers = {"Authorization": f"Token {token}"}

print(f"Fetching project details for {qfield_project_id}...")
p_res = requests.get(f"{baseUrl}/api/v1/projects/{qfield_project_id}/", headers=headers)
if p_res.ok:
    print("Project Details:", p_res.json())
else:
    print(f"Project fetch failed: {p_res.status_code}")

print(f"\nFetching deltas for project {qfield_project_id}...")
d_res = requests.get(f"{baseUrl}/api/v1/deltas/{qfield_project_id}/", headers=headers)
if d_res.ok:
    deltas = d_res.json()
    # Check if results key exists
    if isinstance(deltas, dict) and 'results' in deltas:
        deltas = deltas['results']
        
    print(f"Total deltas found: {len(deltas)}")
    for idx, d in enumerate(deltas):
        print(f"\nDelta {idx + 1}:")
        print(f"  ID: {d.get('id')}")
        print(f"  Status: {d.get('status')}")
        print(f"  Last Status: {d.get('last_status')}")
        print(f"  Created At: {d.get('created_at')}")
        print(f"  Content: {d.get('content')}")
else:
    print(f"Deltas fetch failed: {d_res.status_code} - {d_res.text}")

File: scratch/inspect_deltas.ts
import { PrismaClient } from '@prisma/client';

const prisma = new PrismaClient();
const baseUrl = process.env.NEXT_PUBLIC_QFIELD_API_URL || 'http://localhost:8011';
const projectId = 'cmqpa22sz0000sik8eos6y9o0'; // Database project ID
const qfieldProjectId = '7a55babe-8909-4554-8dd7-9bc03f7a1b9b'; // QField project ID

async function main() {
  console.log("=== Inspecting Local Database ===");
  const project = await prisma.project.findUnique({
    where: { id: projectId },
    select: { name: true, projectCode: true, gisMapping: true }
  });
  console.log("Project info:", project);

  const points = await prisma.surveyPoint.findMany({
    where: { projectId: projectId }
  });
  console.log(`Found ${points.length} survey points in local database.`);
  for (const p of points) {
    console.log(`  - Point ID: ${p.id} | Layer: ${p.layerName} | Status: ${p.verificationStatus}`);
    console.log(`    Attributes:`, JSON.stringify(p.attributes));
  }

  console.log("\n=== Querying QFieldCloud API ===");
  const loginRes = await fetch(`${baseUrl}/api/v1/auth/login/`, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({
      username: process.env.QFIELD_ADMIN_USER || 'admin',
      password: process.env.QFIELD_ADMIN_PASS || 'admin123'
    })
  });

  if (!loginRes.ok) {
    console.error(`Auth failed: ${loginRes.status} - ${await loginRes.text()}`);
    return;
  }

  const { token } = await loginRes.json();
  const headers = { 'Authorization': `Token ${token}` };

  console.log(`Fetching deltas for QField project ${qfieldProjectId}...`);
  const deltasRes = await fetch(`${baseUrl}/api/v1/deltas/${qfieldProjectId}/`, { headers });
  if (!deltasRes.ok) {
    console.error(`Failed to fetch deltas: ${deltasRes.status} - ${await deltasRes.text()}`);
    return;
  }

  const deltasData = await deltasRes.json();
  const deltas = Array.isArray(deltasData) ? deltasData : (deltasData.results || []);
  console.log(`Total QFieldCloud deltas: ${deltas.length}`);

  for (let idx = 0; idx < deltas.length; idx++) {
    const d = deltas[idx];
    console.log(`\nDelta ${idx + 1}:`);
    console.log(`  ID: ${d.id}`);
    console.log(`  Status: ${d.status}`);
    console.log(`  Last Status: ${d.last_status}`);
    console.log(`  Created At: ${d.created_at}`);
    console.log(`  Content:`, JSON.stringify(d.content));
  }
}

main()
  .catch(console.error)
  .finally(() => prisma.$disconnect());
